### PR TITLE
PTX-4744 add AppTaskdown longevity test

### PR DESCRIPTION
**Pull request by thiguetta**
Signed-off-by: Thiago Gonzaga <tgonzaga@purestorage.com>

**What this PR does / why we need it**:
This PR adds AppScaleUpDown to longevity

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: tests/testTriggers.go [hunk 1]
```diff
@@ -3181,6 +3183,113 @@ func getCloudSnapInterval(triggerType string) int {
 
 }
 
+// TriggerAppScaleUpAndDown performs app scale up and down according to chaos level
+func TriggerAppScaleUpAndDown(contexts *[]*scheduler.Context, recordChan *chan *EventRecord) {
```
Comment (vilasd) [inline tests/testTriggers.go:3187]: @thiguetta logic is little convoluted here. Lets talk over a call

File: tests/testTriggers.go [hunk 2]
```diff
@@ -3182,6 +3184,77 @@ func getCloudSnapInterval(triggerType string) int {
...
+				if ok {
+					Step("scaling up apps", func() {
+						applicationScaleUpMap, err := Inst().S.GetScaleFactorMap(ctx)
+						UpdateOutcome(event, err)
+						if err == nil {
+							for key, value := range applicationScaleUpMap {
+								initialScaleFactor[key] = value
+								key = strings.ReplaceAll(key, k8s.DeploymentSuffix, "")
+								key = strings.ReplaceAll(key, k8s.StatefulSetSuffix, "")
+								applicationScaleUpMap[key] = int32(customConfig.CustomPodConfig[key].Replicas)
```
Comment (vilasd) [inline tests/testTriggers.go:3218]: scale down to zero when this event is triggered. and scale up to original replica count at the event interval.

Comment (vilasd) [reply tests/testTriggers.go:3218]: delete all the pods in namespace and wait for them to spin up to exact replicas

File: tests/testTriggers.go [hunk 3]
```diff
@@ -3245,7 +3247,40 @@ func createLongevityJiraIssue(event *EventRecord, err error) {
...
+	chaosLevel := ChaosMap[AppTasksDown]
+	context("deletes all pods from a given app and validate if they recover", func() {
+		for _, ctx := range *contexts {
+			for i := 0; i < chaosLevel; i++ {
+				Step(fmt.Sprintf("delete tasks for app: %s", ctx.App.Key), func() {
+					err := Inst().S.DeleteTasks(ctx, nil)
+					UpdateOutcome(event, err)
+				})
+
+				Step("validate all apps after deletion", func() {
```
Comment (vilasd) [inline tests/testTriggers.go:3277]: @thiguetta will this wait for all the replicas to be teardown and scale up ?

Comment (thiguetta) [reply tests/testTriggers.go:3277]: yes, it will

**Review by vilasd — APPROVED** [review]

**Pull request merged**